### PR TITLE
5.1

### PR DESCRIPTION
**Pull request by jellespijker**
See https://github.com/Ultimaker/Cura/pull/12708 for the full overview

### CODE REVIEW

**Review by Joeydelarago — APPROVED** [review]

File: GitVersion.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        increment: None
+        prevent-increment-of-merged-branch-version: false
+        track-merge-target: true
+        source-branches: [ 'main' ]
+        tracks-release-branches: true
+        is-release-branch: false
+        is-mainline: false
+        pre-release-weight: 0
+    release:
+        regex: ^[\d].[\d]$
```
Comment (casperlamboo) [inline GitVersion.yml:28]:
```suggestion
        regex: ^[0-9]+.[0-9]+$
```

File: .github/workflows/conan-package.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            - 'CMakeLists.txt'
+            - '.github/workflows/conan-package.yml'
+            - '.github/workflows/requirements*'
+            - 'GitVersion.yml'
+            - 'conandata.yml'
+            - '.gitattributes'
+        branches:
+            - main
+            - 'CURA-*'
+            - '[1-9]+.[0-9]+'
```
Comment (casperlamboo) [inline .github/workflows/conan-package.yml:43]:
```suggestion
            - '[1-9]+.[0-9]+'
```

File: .github/workflows/conan-package.yml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            - main
+            - 'CURA-*'
+            - '[0-9]+.[0-9]+'
+        tags:
+            - '[0-9]+.[0-9]+.[0-9]+'
+
+
+jobs:
+    conan-recipe-version:
+        uses: ultimaker/cura/.github/workflows/conan-recipe-version.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:50]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-recipe-version.yml@main
```

File: .github/workflows/conan-package.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        with:
+            project_name: savitar
+
+    conan-package-export:
+        needs: [ conan-recipe-version ]
+        uses: ultimaker/cura/.github/workflows/conan-recipe-export.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:56]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-recipe-export.yml@main
```

File: .github/workflows/conan-package.yml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            recipe_id_pr: ${{ needs.conan-recipe-version.outputs.recipe_id_pr }}
+            runs_on: 'ubuntu-20.04'
+            python_version: '3.10.4'
+            conan_config_branch: 'master'
+            conan_logging_level: 'info'
+        secrets: inherit
+
+    conan-package-create-macos:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ (github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_macos) }}
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:69]:
```suggestion
        if: ${{ (github.event_name == 'push' && github.ref_name == 'main')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_macos) }}
```

File: .github/workflows/conan-package.yml [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        needs: [ conan-recipe-version, conan-package-export ]
+
+        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:72]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@main
```

File: .github/workflows/conan-package.yml [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+            recipe_id_full: ${{ needs.conan-recipe-version.outputs.recipe_id_full }}
+            runs_on: 'macos-10.15'
+            python_version: '3.10.4'
+            conan_config_branch: 'master'
+            conan_logging_level: 'info'
+        secrets: inherit
+
+    conan-package-create-windows:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ (github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_windows) }}
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:83]:
```suggestion
        if: ${{ (github.event_name == 'push' && github.ref_name == 'main') || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_windows) }}
```

File: .github/workflows/conan-package.yml [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        needs: [ conan-recipe-version, conan-package-export ]
+
+        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:86]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@main
```

File: .github/workflows/conan-package.yml [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+            recipe_id_full: ${{ needs.conan-recipe-version.outputs.recipe_id_full }}
+            runs_on: 'windows-2022'
+            python_version: '3.10.4'
+            conan_config_branch: 'master'
+            conan_logging_level: 'info'
+        secrets: inherit
+
+    conan-package-create-linux:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ (github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux) }}
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:97]:
```suggestion
        if: ${{ (github.event_name == 'push' && github.ref_name == 'main') || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux) }}
```

File: .github/workflows/conan-package.yml [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+        needs: [ conan-recipe-version, conan-package-export ]
+
+        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:100]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@main
```

File: .github/workflows/conan-package.yml [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+            python_version: '3.10.4'
+            conan_config_branch: 'master'
+            conan_logging_level: 'info'
+        secrets: inherit
+
+    notify-export:
+        if: ${{ always() }}
+        needs: [ conan-package-export ]
+
+        uses: ultimaker/cura/.github/workflows/notify.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:113]:
```suggestion
        uses: ultimaker/cura/.github/workflows/notify.yml@main
```

File: .github/workflows/conan-package.yml [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+            success: ${{ contains(join(needs.*.result, ','), 'success') }}
+            success_title: "New Conan recipe exported in ${{ github.repository }}"
+            success_body: "Exported ${{ needs.conan-recipe-version.outputs.recipe_id_full }}"
+            failure_title: "Failed to export Conan Export in ${{ github.repository }}"
+            failure_body: "Failed to exported ${{ needs.conan-recipe-version.outputs.recipe_id_full }}"
+        secrets: inherit
+
+    notify-create:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ always() && ((github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux)) }}
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:124]:
```suggestion
        if: ${{ always() && ((github.event_name == 'push' && github.ref_name == 'main') || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux)) }}
```

File: .github/workflows/conan-package.yml [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+        needs: [ conan-package-create-macos, conan-package-create-windows, conan-package-create-linux ]
+
+        uses: ultimaker/cura/.github/workflows/notify.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:127]:
```suggestion
        uses: ultimaker/cura/.github/workflows/notify.yml@main
```

**Review by Joeydelarago — COMMENTED** [review]
Update branch names

File: .github/workflows/conan-package.yml (hunk 3 above)
Comment (casperlamboo) [inline .github/workflows/conan-package.yml:50]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-recipe-version.yml@main
```

**Pull request merged**